api/replica.go

```diff
@@ -14,20 +19,66 @@ type Replica struct {
 	handler   http.Handler
 	tlsConfig *tls.Config
 	primary   string
+	addr      string // the address of this replica
+	// mu is the mutex used for locking the primary.
+	mu sync.RWMutex
```

Comment:
nit: add 'primary' prefix to make it consistent with 'primaryWait' below ?

Reply:
I believe it's convention to use `mu` since it's clear what it's doing based on who owns it.